doc/src/tutorial/index.rst

```diff
@@ -17,3 +17,4 @@
    solvers.rst
    matrices.rst
    manipulation.rst
+   codegen.rst
```

Comment:
Do we need an advanced tutorial?

This doesn't really go with any particular module because the modules in it are spread around. It could be listed as a general topic instead of a tutorial. It is pretty much like a tutorial but it borders on just being an explanation of the modules too. But this can also be considered the beginner's tutorial for the code generation aspects of SymPy. I feel like the code gen stuff is more related to the core of sympy functionality than other extra modules, like mechanics, etc.